### PR TITLE
Add SOC & multiple `PROCAR` parsing functionalities

### PR DESCRIPTION
**Pull request by kavanase**
This PR implements the `PROCAR` parsing functionality from [`easyunfold`](https://smtg-bham.github.io/easyunfold/) discussed in #3722; namely the ability to parse `PROCAR`s from spin-orbit coupling (SOC) VASP calculations (i.e. with `LSORBIT = True`), and the ability to parse multiple `PROCAR` files together (without wasting time parsing duplicate k-points etc; e.g. if the user has performed split hybrid DFT band structure calculations).

Also adds some further attributes to `Procar`; `is_soc`, `kpoints`, `occupancies`, `eigenvalues` and (when `is_soc` is `True`) `xyz_data` which gives the projections data along the x/y/z magnetisation projections.

Tests added for parsing SOC `PROCAR`s, and parsing multiple `PROCAR`s – including when they are SOC and `LORBIT = 14` (so phase factors also included and being parsed correctly).

### CODE REVIEW

File: pymatgen/io/vasp/outputs.py [hunk 1]
```diff
@@ -3799,59 +3805,258 @@ class Procar:
...
+            eigenvalues_list.append(eigenvalues)
+            data_list.append(data)
+            xyz_data_list.append(xyz_data)
+            phase_factors_list.append(phase_factors)
+
+        # Combine arrays along the kpoints axis:
+        # nbands (axis = 2) could differ between arrays, so set missing values to zero:
+        max_nbands = max(eig_dict[Spin.up].shape[1] for eig_dict in eigenvalues_list)
+        for dict_array_list in [occupancies_list, eigenvalues_list, data_list, xyz_data_list, phase_factors_list]:
+            for dict_array in dict_array_list:
```
Comment (shyuep) [inline pymatgen/io/vasp/outputs.py:3862]: I think itertools.chain would work far better here?

Comment (kavanase) [reply pymatgen/io/vasp/outputs.py]: Ah yes good point, done! Doesn't make a different to efficiency as limiting factor by far is IO, but cleaner!

**Comment by mkhorton** [comment]
Thanks for the PR @kavanase! It looks like we have a merge conflict before we can merge, otherwise this looks great.

**Comment by kavanase** [comment]
No worries! Yep merge conflict came from some updates in the master branch since opening this, resolved now!

**Comment by kavanase** [comment]
Just to bump, I think this PR is all ready to go. Fixed the merge conflict that had been introduced with the updated `master` branch

**Review by janosh — APPROVED** [review]
thanks @kavanase! 👍

**Pull request merged**